examples/pytorch/domain_templates/generative_adversarial_net.py

```diff
@@ -59,7 +58,7 @@ def block(in_feat, out_feat, normalize=True):
             *block(128, 256),
             *block(256, 512),
             *block(512, 1024),
-            nn.Linear(1024, int(np.prod(img_shape))),
+            nn.Linear(1024, int(torch.prod(img_shape))),
```

Comment:
`torch.prod` only works on tensors, unlike the previous numpy call which works on iterables.
So we could use `math.prod` here as a replacement
```suggestion
            nn.Linear(1024, int(math.prod(img_shape))),
```
(and add missing math import)